readthedocs/core/utils/extend.py

```diff
@@ -0,0 +1,51 @@
+"""Patterns for extending Read the Docs"""
+
+from django.conf import settings
+from django.utils.module_loading import import_by_path
+from django.utils.functional import LazyObject
+
+
+class SettingsOverrideObject(LazyObject):
```

Comment:
Would be good to see a test of this class -- mainly just to wrap my head around the usage. I guess most of the logic is enabled by Django's `LazyObject`, but might be good to at least test our integration.